src/warehouse/Domain/Entities/ProductPurchase.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Domain.Entities
+{
+    public class ProductPurchase
+    {
+        public Guid Id { get; set; }
+        public string ProductName { get; set; }
+        public string ProductUom { get; set; }
+        public decimal Price { get; set; }
+        public Guid WareHouseId { get; set; }
+        public int Quantity { get; set; }
+        public string VendorName { get; set; }
+        public decimal TotalPrice { get; set; }
+        public DateTime Date { get; set; }
+        public string Comment { get; set; }
+        [Timestamp]
```

Comment:
Can you create test for that?

Reply:
I think just adding ConcurrencyCheck attribute on Quantity is enough: https://www.learnentityframeworkcore.com/configuration/data-annotation-attributes/concurrencycheck-attribute#:~:text=EF%20Core%20ConcurrencyCheck%20Attribute,as%20part%20of%20concurrency%20management.